File: ios/Swapl/Core/Repositories/VerificationRepository.swift
import Foundation

// Didit identity verification (hosted flow).
//
// The backend is env-gated: with Didit unconfigured, /status still answers
// (enabled=false → the client hides the CTA) while /session returns 503.

// GET /api/verification/status
struct VerificationStatus: Decodable, Sendable {
    let enabled: Bool
    /// "none" | "pending" | "approved" | "declined" | ...
    let status: String
    let verified: Bool
    let verifiedAt: String?
    let completedAt: String?
    /// Present only when this now-verified user was referred and the two-sided
    /// referral Keys reward paid out — drives the post-verify "you earned Keys"
    /// toast. Derived from persisted state, so it's stable across polls.
    let referralReward: ReferralReward?
}

// Referee-side reward surfaced by /api/verification/status after the invitee
// verifies and the referral qualifies.
struct ReferralReward: Decodable, Sendable {
    let keys: Int
    let referrerName: String?
}

// POST /api/verification/session — `url` is the hosted Didit page (nil once
// already approved).
struct VerificationSessionStart: Decodable, Sendable {
    let status: String
    let url: String?
}

final class VerificationRepository: @unchecked Sendable {
    static let shared = VerificationRepository()

    func status() async throws -> VerificationStatus {
        try await APIClient.shared.send("GET", "/api/verification/status")
    }

    func createSession() async throws -> VerificationSessionStart {
        try await APIClient.shared.send("POST", "/api/verification/session")
    }
}

File: android/swapl/app/src/main/java/app/swapl/core/repository/VerificationRepository.kt
package app.swapl.core.repository

import app.swapl.core.network.ApiClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import io.ktor.client.request.post
import javax.inject.Inject
import javax.inject.Singleton

// Didit identity verification (hosted flow), mirroring the iOS repository.
//
// The backend is env-gated: with Didit unconfigured, /status still answers
// (enabled=false → the client hides the CTA) while /session returns 503.

// GET /api/verification/status
@kotlinx.serialization.Serializable
data class VerificationStatus(
    val enabled: Boolean = false,
    /** "none" | "pending" | "approved" | "declined" | ... */
    val status: String = "none",
    val verified: Boolean = false,
    val verifiedAt: String? = null,
    val completedAt: String? = null,
    /**
     * Present only when this now-verified user was referred and the two-sided
     * referral Keys reward paid out — drives the post-verify "you earned Keys"
     * toast. Derived from persisted state, stable across polls.
     */
    val referralReward: ReferralReward? = null,
)

// Referee-side reward surfaced by /api/verification/status after the invitee
// verifies and the referral qualifies.
@kotlinx.serialization.Serializable
data class ReferralReward(
    val keys: Int = 0,
    val referrerName: String? = null,
)

// POST /api/verification/session — `url` is the hosted Didit page (null once
// already approved).
@kotlinx.serialization.Serializable
data class VerificationSessionStart(
    val status: String,
    val url: String? = null,
)

@Singleton
class VerificationRepository @Inject constructor(private val api: ApiClient) {

    suspend fun status(): VerificationStatus =
        api.client.get("${api.baseUrl}/api/verification/status").body()

    suspend fun createSession(): VerificationSessionStart =
        api.client.post("${api.baseUrl}/api/verification/session").body()
}
